src/Data/List/Relation/Binary/Permutation/Inductive.agda

```diff
@@ -62,11 +62,22 @@ data _↭_ : Rel (List A) a where
 module PermutationReasoning where
 
   open EqReasoning ↭-setoid
-    using (_IsRelatedTo_; relTo)
+    using (_IsRelatedTo_; relTo; step-≈; step-≈˘)
 
   open EqReasoning ↭-setoid public
-    using (begin_ ; _∎ ; _≡⟨⟩_; _≡⟨_⟩_)
-    renaming (_≈⟨_⟩_ to _↭⟨_⟩_; _≈˘⟨_⟩_ to _↭˘⟨_⟩_)
```

Comment:
Why do you need to redefine these rather than just renaming them as before?

Reply:
I didn’t think that was possible (I ran into errors when I tried to do `renaming` with the `syntax` declarations). And when I renamed the `step-` combinators, the new `syntax` declarations didn’t work.

Reply:
Ah yes because the syntax is defined with arguments in place! We really don't want to have to redefine all the primitives every time we want to rename the combinators. Let me think...

Reply:
Any intuition about whether we'd still get the performance increase if we made them definitions instead of syntax? Obviously there'd be a small unfolding hit, but would we still get the inference benefit? Pinging @nad

Reply:
I don't think that would work! The interface should be:

```agda
x + y
  ={ +-comm x y }
y + x
```

So if it's a definition the type checker will go:

1. x + y
2. +-comm x y
3. y + x

AFAIK, the only way to make the interface be ordered 1, 2, 3 but the *typechecking* to go 1, 3, 2 is with a syntax declaration

.